File: CS/NavBarExample.Module/BusinessObjects/DomainObjects.cs
using System;
using System.ComponentModel;

using DevExpress.Xpo;
using DevExpress.Data.Filtering;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace NavBarExample.Module.BusinessObjects {
    [NavigationItem("Group1")]
    public class DomainObject1 : BaseObject {
        public DomainObject1(Session session)
            : base(session) { }
    }
    [NavigationItem("Group1")]
    public class DomainObject2 : BaseObject {
        public DomainObject2(Session session)
            : base(session) { }
    }
    [NavigationItem("Group1")]
    public class DomainObject3 : BaseObject {
        public DomainObject3(Session session)
            : base(session) { }
    }
    [NavigationItem("Group2")]
    public class DomainObject4 : BaseObject {
        public DomainObject4(Session session)
            : base(session) { }
    }
    [NavigationItem("Group2")]
    public class DomainObject5 : BaseObject {
        public DomainObject5(Session session)
            : base(session) { }
    }
    [NavigationItem("Group2")]
    public class DomainObject6 : BaseObject {
        public DomainObject6(Session session)
            : base(session) { }
    }
    [NavigationItem("Group3")]
    public class DomainObject7 : BaseObject {
        public DomainObject7(Session session)
            : base(session) { }
    }
}

File: VB/NavBarExample.Module/BusinessObjects/DomainObjects.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel

Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace NavBarExample.Module.BusinessObjects
	<NavigationItem("Group1")> _
	Public Class DomainObject1
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
	End Class
	<NavigationItem("Group1")> _
	Public Class DomainObject2
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
	End Class
	<NavigationItem("Group1")> _
	Public Class DomainObject3
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
	End Class
	<NavigationItem("Group2")> _
	Public Class DomainObject4
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
	End Class
	<NavigationItem("Group2")> _
	Public Class DomainObject5
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
	End Class
	<NavigationItem("Group2")> _
	Public Class DomainObject6
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
	End Class
	<NavigationItem("Group3")> _
	Public Class DomainObject7
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
	End Class
End Namespace
